### ISSUE TITLE
False positives on container attributes Pydantic v2

### ISSUE THREAD

**Issue by kamichal** [pylint-dev/pylint#10087]
### Bug description

```python
E1101: Instance of 'FieldInfo' has no 'append' member (no-member)
and similar bugs, when pylint thinks the type of the attribute a1 below is an instance of FieldInfo

## Reproduction code


from pydantic import BaseModel, Field


class A(BaseModel):
    a1: list = Field(default_factory=list)
    a2: dict = Field(default_factory=dict)

    def do(self):
        self.a1.append(8)
        self.a2[5] = 6


obj = A(a1=[12])
obj.a1.append(13)
```

### Configuration

```ini

```

### Command used

```shell
pylint scratches/pylint_bug.py
```

### Pylint output

```python
pylint_bug.py:9:8: E1101: Instance of 'FieldInfo' has no 'append' member (no-member)
pylint_bug.py:10:8: E1137: 'self.a2' does not support item assignment (unsupported-assignment-operation)
pylint_bug.py:14:0: E1101: Instance of 'FieldInfo' has no 'append' member (no-member)
```

### Expected behavior

Pylint should use the type defined by `default_factory` argument of `Field`.

### Pylint version

```shell
pydantic==2.9.2
Python 3.11.9
```

### OS / Environment

Linux 6.6.54-2-MANJARO x86_64 GNU/Linux

### Additional dependencies

```python
pylint==3.3.1
```
